config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -416,6 +440,432 @@ type MetricsServerConfig struct {
+	// action is the action to perform.
+	// Valid actions are: replace, keep, drop, hashmod, labelmap, labeldrop, labelkeep.
+	// +required
+	Action RelabelAction `json:"action,omitempty"`
+}
+
+// TLSConfig represents TLS configuration for Alertmanager connections.
+type TLSConfig struct {
+	// ca is the CA certificate to use for TLS connections.
+	// +optional
+	CA SecretKeySelector `json:"ca,omitempty,omitzero"`
+	// cert is the client certificate to use for TLS connections.
+	// +optional
+	Cert SecretKeySelector `json:"cert,omitempty,omitzero"`
+	// key is the client key to use for TLS connections.
+	// +optional
+	Key SecretKeySelector `json:"key,omitempty,omitzero"`
+	// serverName is the server name to use for TLS connections.
+	// If specified, must be a valid DNS subdomain as per RFC 1123.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	ServerName string `json:"serverName,omitempty"`
+	// certificateVerification determines the policy for TLS certificate verification.
+	// Allowed values are "Verify" (performs certificate verification, secure) and "SkipVerify" (skips verification, insecure).
+	// When omitted, defaults to "Verify" (secure certificate verification is performed).
+	// +optional
+	// +kubebuilder:validation:Enum=Verify;SkipVerify
+	// +kubebuilder:default=Verify
+	CertificateVerification string `json:"certificateVerification,omitempty"`
+}
+
+// SecretKeySelector selects a key of a Secret.
+// +structType=atomic
+type SecretKeySelector struct {
+	// name is the name of the secret in the same namespace to select from.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
+	// key is the key of the secret to select from. Must be a valid secret key.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Key string `json:"key,omitempty"`
+	// optional specifies whether the Secret or its key must be defined.
+	// When set to "Required", the Secret and key must exist; if missing, the operator will report
+	// an error condition and block reconciliation until the secret is created.
+	// When set to "Optional", the CRD validation will pass even if the Secret or key is missing.
+	// The operator will continue reconciliation with reduced functionality:
+	//   - Authentication or TLS credentials dependent on this secret will be skipped.
+	//   - Prometheus will attempt to communicate without the missing credentials, which may result
+	//     in failed connections or unauthenticated access depending on the remote endpoint's requirements.
+	//   - The operator may emit warning events indicating the missing secret.
+	//   - No automatic fallback values are applied; the secret reference is simply ignored.
+	// When omitted, defaults to "Required".
+	// +optional
+	// +kubebuilder:default=Required
+	Optional SecretRequirement `json:"optional,omitempty"`
```

Comment:
I think that we can make all secret keys required and not have the optional field. The Prometheus operator itself will fail the reconciliation if a secret key ref is missing hence a ref is required de facto.